### PR TITLE
Fog probability

### PR DESCRIPTION
**Pull request by GlennGoddard**
Add Fog Probability
This is a prerequisite for local current conditions
I did simplify calculated fog probability to not include temperature trend data since that is not readily available but it does not have a huge effect not utilizing temperature trends.

Snow Probability will be next and then should have all requirement for current conditions.

I think I got all the appropriate files updated, but I have a nagging feeling that I forgot something.

### CODE REVIEW

**Comment by briis** [comment]
Thanks Glenn,
I will merge and test it out.

File: weatherflow2mqtt/helpers.py [hunk 1]
```diff
@@ -913,3 +913,78 @@ def zambretti_forecast(self, z_num: int):
...
+        if (
+            solar_elevation is None
+            or wind_speed is None
+            or humidity is None
+            or dew_point is None
+        ):
+            return None
+        
+        fog = 0
+        diff = temperature - dew_point
```
Comment (briis) [inline weatherflow2mqtt/helpers.py:935]: Where is this temperature value coming from?

**Comment by briis** [comment]
Hi Glenn,
I had to make some modifications, but there is now a `dev` container which creates this sensor. Can you have a run down on this, and see if this produces the right sensor and the right value?

File: weatherflow2mqtt/helpers.py (hunk 1 above)
Comment (briis) [reply weatherflow2mqtt/helpers.py:935]: I decided that this was the current air temperature, and supplied that to the formula. If this is not correct, let me know.

**Comment by briis** [comment]
When done, please also make an update to the README file.
Is this btw the probability for fog for the next 24 hours or what is the timeframe?

**Comment by GlennGoddard** [comment]
Yes, temperature is air temperature (I'm not sure how I missed that since I copied and pasted from my running script, then converted from java-script to pyhton).
This is current conditions for fog; although for my personal testing when at 27% fog probability there will be fog on my way to work usually within a few miles of leaving my home.

I will update the readme and test the dev build (I will be about 10 hours before I can do this).  Hopefully, I can get the Snow Probability to you before the end of your weekend.

**Comment by GlennGoddard** [comment]
I updated ReadMe and will do that PR once I update to the newest DEV build and verify everything.

**Comment by GlennGoddard** [comment]
@briis It is off by a lot when conditions are right.  I need to see what the issue is.  Both calculations were zero (0) until the same time and then started diverging.

**Comment by briis** [comment]
OK, I let you dig around. 😁

**Pull request merged**